source/fundamentals/poco.txt

```diff
@@ -0,0 +1,249 @@
+.. _csharp-poco:
+
+===============
+Work with POCOs
+===============
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn about how you can use "Plain Old CLR/Class
+Objects", or **POCOs**, with the {+driver-short+} for your operations
+and queries. POCOs are simple class objects which do not inherit
+features from any framework-specific base class and do not return any
+special types. We recommend using POCOs in your {+language+} code to
+adhere to idiomatic driver usage and achieve the best performance.
+
+You should read this guide if you want to learn more about how to use
+POCOs with the {+driver-short+} or need to adjust the driver's default
+field mapping behavior.
+
+Create a POCO
+-------------
+
+You can create a POCO by instantiating any simple class that does not
+implement interfaces or extend classes from a framework. When you
+execute an operation such as a read or write using a POCO, the driver
+interally *serializes*, or converts, the POCO to BSON.
+
+Select the :guilabel:`POCO` or :guilabel:`BSON` tab to see how the
+driver serializes a sample POCO to BSON:
+
+.. tabs::
+
+   .. tab:: POCO
+      :tabid: poco-representation
+
+      .. code-block:: csharp
+         :copyable: false
+
+         public class Clothing
+         {
+             public ObjectId _id { get; set; }
+             public string name { get; set; }
+             public bool instock { get; set; }
+             public double price { get; set; }
+             public List<string> color { get; set; }
```

Comment:
If I use the C# naming conventions in the class, then when they are serialized to BSON, the fieldnames will also be capitalized/in PascalCase. Is that the tradeoff I should make? I'm not sure what you mean by adding a note.

Reply:
edit-- I just kept PascalCase to follow C# and let the BSON also be in PascalCase. Instead i added the note about creating a `ConventionPack`. I think its better to follow the C# conventions, because any BSON conventions are not as strict.

Reply:
Sorry to be vague. I meant that you could keep the unconventional property names here (because, without `ConventionPack` or a Bson attribute, that's what's required for BSON serialization to work). We could then call out the fact that even though this example works, we can do better by correctly capitalizing the properties. 

In the **Set Field Names** section, what do you think about mentioning that BSON fields in the first example were PascalCase to match the C#, but should generally be camelCase--and that's the advantage of using BsonElement/ConventionPack?